### PR TITLE
Show pool/volume informations where possible

### PR DESCRIPTION
**Pull request by cbosdo**
## What does this PR change?

At least for Xen the disks of volume types aren't supported but
converted by Salt into their file or block equivalents. In order to show
the selected volume and pool when possible, try to get those
informations from Salt's virt.vm_info output.

## GUI diff

No difference.

- [X] **DONE**

## Documentation
- No documentation needed: fixes VM disks infos for Xen

- [X] **DONE**

## Test coverage
- Unit tests were updated

- [X] **DONE**

## Links

- [X] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_lint_checkstyle"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Comment by cbosdo** [comment]
Requires https://github.com/openSUSE/salt/pull/264 for the testsuite

**Review by mateiw — CHANGES_REQUESTED** [review]

File: java/code/src/com/suse/manager/virtualization/GuestDiskDef.java [hunk 1]
```diff
@@ -216,6 +218,16 @@ else if ("cdrom".equals(disk.getDevice())) {
                         source.put("file", info.getFile());
                     }
                 }
+                else if (Arrays.asList("block", "file").contains(disk.getType())) {
+                    Pattern regex = Pattern.compile("([^/]+)/([^/]+)");
```
[the remark this answers is not in the document]
Comment (cbosdo) [reply java/code/src/com/suse/manager/virtualization/GuestDiskDef.java]: done. Thanks for the suggestion

**Review by mateiw — APPROVED** [review]
LGTM :+1:

File: testsuite/features/secondary/minkvm_guests.feature [hunk 2]
```diff
@@ -129,7 +129,7 @@ Feature: Be able to manage KVM virtual machines via the GUI
     And "test-vm" virtual machine on "kvm_server" should have spice graphics device
     And "test-vm" virtual machine on "kvm_server" should have 1 NIC using "test-net1" network
     And "test-vm" virtual machine on "kvm_server" should have a NIC with 02:34:56:78:9a:bc MAC address
-    And "test-vm" virtual machine on "kvm_server" should have a "test-vm_disk.qcow2" scsi disk
+    And "test-vm" virtual machine on "kvm_server" should have a "test-vm_disk.qcow2" scsi disk from pool "tmp"
```
Comment (Bischoff) [inline testsuite/features/secondary/minkvm_guests.feature:132]: scsi is an abbreviation: SCSI

File: testsuite/features/secondary/minxen_guests.feature [hunk 3]
```diff
@@ -192,21 +186,21 @@ Feature: Be able to manage XEN virtual machines via the GUI
...
+    And "test-vm2" virtual machine on "xen_server" should have a "/var/lib/libvirt/images/test-pool0/test-vm2_system" xen disk
 
 @virthost_xen
-  Scenario: Show the Spice graphical console for Xen
+  Scenario: Show the VNC graphical console for Xen
     Given I am on the "Virtualization" page of this "xen_server"
     When I click on "Graphical Console" in row "test-vm2"
     And I switch to last opened window
-    Then I wait until I see the spice graphical console
+    Then I wait until I see the VNC graphical console
```
Comment (Bischoff) [inline testsuite/features/secondary/minxen_guests.feature:203]: could be "And"

File: testsuite/features/secondary/minxen_guests.feature [hunk 4]
```diff
@@ -218,13 +212,21 @@ Feature: Be able to manage XEN virtual machines via the GUI
     And I enter "512" as "memory"
     And I enter "/var/testsuite-data/disk-image-template.qcow2" as "disk0_source_template"
     And I select "test-net0" from "network0_source"
+    And I select "Spice" from "graphicsType"
     And I click on "Create"
     Then I should see a "Hosted Virtual Systems" text
     When I wait until I see "test-vm3" text
     And I wait at most 500 seconds until table row for "test-vm3" contains button "Stop"
     And "test-vm3" virtual machine on "xen_server" should have 512MB memory and 1 vcpus
```
Comment (Bischoff) [inline testsuite/features/secondary/minxen_guests.feature:220]: "Then"

File: testsuite/features/secondary/minxen_guests.feature [hunk 5]
```diff
@@ -218,13 +212,21 @@ Feature: Be able to manage XEN virtual machines via the GUI
...
     And "test-vm3" virtual machine on "xen_server" should have 1 NIC using "test-net0" network
-    And "test-vm3" virtual machine on "xen_server" should have a "test-vm3_system" xen disk from pool "test-pool0"
+    And "test-vm3" virtual machine on "xen_server" should have a "/var/lib/libvirt/images/test-pool0/test-vm3_system" xen disk
+
+@virthost_xen
+  Scenario: Show the Spice graphical console for Xen
+    Given I am on the "Virtualization" page of this "xen_server"
+    When I click on "Graphical Console" in row "test-vm3"
+    And I switch to last opened window
+    Then I wait until I see the spice graphical console
```
Comment (Bischoff) [inline testsuite/features/secondary/minxen_guests.feature:229]: Could be "And"

**Review by Bischoff — CHANGES_REQUESTED** [review]
See nitpicks.

File: testsuite/features/secondary/minkvm_guests.feature (hunk 2 above)
Comment (cbosdo) [reply testsuite/features/secondary/minkvm_guests.feature:132]: OK, I'll lowercase it in the step then

File: testsuite/features/secondary/minxen_guests.feature (hunk 3 above)
Comment (cbosdo) [reply testsuite/features/secondary/minxen_guests.feature:203]: OK

File: testsuite/features/secondary/minxen_guests.feature (hunk 4 above)
Comment (cbosdo) [reply testsuite/features/secondary/minxen_guests.feature:220]: That one is not new ;)

File: testsuite/features/secondary/minxen_guests.feature (hunk 5 above)
Comment (srbarrios) [reply testsuite/features/secondary/minxen_guests.feature:229]: Actually, I prefer it as Then :D I think we agreed to don't consider the "I wait until" part, sometime in the past.

**Comment by cbosdo** [comment]
@Bischoff fixed

**Pull request merged**